packages/edge-login-ui-rn/src/components/themed/EdgeTextFieldOutlined.js

```diff
@@ -121,291 +146,288 @@ const CornerBorder = ({
+
+  // animation
+  const inputRef = useRef<TextInput>(null)
+  const placeholderMap = useSharedValue(
+    value ? ANIMATION_STATES.FOCUSED : ANIMATION_STATES.INIT
+  )
+  const placeholderSize = useSharedValue(ANIMATION_STATES.INIT)
+  const containerWidth = useSharedValue(ANIMATION_STATES.INIT)
+  const colorMap = useSharedValue(ANIMATION_STATES.INIT)
+
+  // input methods
+  const focus = () => inputRef.current && inputRef.current.focus()
+  const blur = () => inputRef.current && inputRef.current.blur()
+  const isFocused = () =>
+    Boolean(inputRef.current && inputRef.current.isFocused())
+  const clear = () => {
+    Boolean(inputRef.current && inputRef.current.clear())
+    if (onChangeText != null) onChangeText('')
+  }
 
-    // styles
-    const theme = useTheme()
-    const styles = getStyles(theme)
-    const spacings = sidesToMargin(
-      mapSides(fixSides(marginRem, 0.5), theme.rem)
-    )
-    const placeholderSpacerWidthAdjust = theme.rem(1)
-    const {
-      inactiveColor,
-      activeColor,
-      errorColor,
-      fontSize,
-      placeholderSpacerAdjust,
-      placeholderScale,
-      placeholderSizeScale,
-      paddingVertical,
-      inputStyles,
-      placeholderTextStyles,
-      placeholderPaddingStyles,
-      placeholderSpacerPaddingStyles,
-      inputContainerStyles,
-      prefixStyles,
-      suffixStyles,
-      hintLeftMargin
-    } = getSizeStyles(theme, size, showSearchIcon)
-
-    const errorState = useCallback(
-      () => error !== null && error !== undefined,
-      [error]
-    )
+  // styles
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const spacings = sidesToMargin(mapSides(fixSides(marginRem, 0.5), theme.rem))
+  const placeholderSpacerWidthAdjust = theme.rem(1)
+  const {
+    inactiveColor,
+    activeColor,
+    errorColor,
+    fontSize,
+    placeholderSpacerAdjust,
+    placeholderScale,
+    placeholderSizeScale,
+    paddingVertical,
+    inputStyles,
+    placeholderTextStyles,
+    placeholderPaddingStyles,
+    placeholderSpacerPaddingStyles,
+    inputContainerStyles,
+    prefixStyles,
+    suffixStyles,
+    hintLeftMargin
+  } = getSizeStyles(theme, size, showSearchIcon)
 
-    const handleFocus = e => {
-      placeholderMap.value = withTiming(ANIMATION_STATES.FOCUSED)
-      if (!errorState()) colorMap.value = withTiming(ANIMATION_STATES.FOCUSED)
-      focus()
-      if (onFocus) onFocus(e)
-    }
+  const errorState = useCallback(() => error != null, [error])
 
-    const handleBlur = e => {
-      if (!value) placeholderMap.value = withTiming(ANIMATION_STATES.INIT) // blur
-      if (!errorState()) colorMap.value = withTiming(ANIMATION_STATES.INIT) // inactive
-      blur()
-      if (onBlur) onBlur(e)
-    }
+  const handleFocus = () => {
+    placeholderMap.value = withTiming(ANIMATION_STATES.FOCUSED)
+    if (!errorState()) colorMap.value = withTiming(ANIMATION_STATES.FOCUSED)
+    focus()
+    if (onFocus != null) onFocus()
```

Comment:
Will removing the error object from propagating through be a breaking change for edge-login-ui?

Reply:
No, nobody uses it.